security/pkg/server/ca/server.go

```diff
@@ -76,10 +77,16 @@ func (s SaNode) String() string {
 func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertificateRequest) (
 	*pb.IstioCertificateResponse, error,
 ) {
+	peerAddr := "0.0.0.0"
+	if peerInfo, ok := peer.FromContext(ctx); ok {
+		peerAddr = peerInfo.Addr.String()
```

Comment:
I am afraid the peer address may not be original address, because of nat or lb in front.

Reply:
I just want to know which client failed to request the CSR. I referred to the authentication logic of the discovery server for this part.
https://github.com/istio/istio/blob/47f6d3dbdbe504ea4967d9665322d63ee8126235/pilot/pkg/xds/auth.go#L41-L66

Indeed, there might be cases where NAT or an SLB is involved. However, I believe that printing the client source can be helpful in quickly trouble shooting in many situations.

Reply:
fair